Core/WebCacheManager.swift

```diff
@@ -139,30 +151,124 @@ public class WebCacheManager {
 
     }
 
-    public func clear(dataStore: WebCacheManagerDataStore = WKWebsiteDataStore.default(),
-                      logins: PreserveLogins = PreserveLogins.shared,
+    @available(iOS 17, *)
+    func checkDataStores() async {
+        let ids = await WKWebsiteDataStore.allDataStoreIdentifiers
+        if ids.count >  0 {
+            Pixel.fire(pixel: .debugUnexpectedWebsiteDataStores)
+            assertionFailure("Unexpected number of data stores")
+        }
+    }
+
+    @available(iOS 17, *)
+    func containerBasedClearing(logins: PreserveLogins,
+                                storeIdManager: DataStoreIdManager,
+                                completion: @escaping () -> Void) {
+
+        guard let containerId = storeIdManager.id else {
+            completion()
+            return
+        }
+
+        Task { @MainActor in
+            var dataStore: WKWebsiteDataStore? = WKWebsiteDataStore(forIdentifier: containerId)
+            let cookies = await dataStore?.preservedCookies(logins)
+            dataStore = nil
+
+            let uuids = await WKWebsiteDataStore.allDataStoreIdentifiers
+            for uuid in uuids {
+                do {
+                    try await WKWebsiteDataStore.remove(forIdentifier: uuid)
+                } catch {
+                    Pixel.fire(pixel: .debugCouldNotRemoveWebsiteDataStore, error: error)
+
+                    // Not a fatal error because it will get deleted next time
+                    assertionFailure("Failed to remove datastore")
```

Comment:
During testing on simulator I was hit it with the following error:
`Error Domain=WKWebSiteDataStore Code=1 "Data store is in use (by network process)" UserInfo={NSLocalizedDescription=Data store is in use (by network process)}`

Reply:
And: `Error Domain=WKWebSiteDataStore Code=1 "Data store is in use" UserInfo={NSLocalizedDescription=Data store is in use}`